### PR TITLE
all: yet another Soroban XDR bump

### PR DESCRIPTION
**Pull request by 2opremio**
Based on [ec04dce3983103f8ed8b3b25ef028c08c9d8c59d](https://github.com/stellar/stellar-xdr/tree/ec04dce3983103f8ed8b3b25ef028c08c9d8c59d)

Fixes part of https://github.com/stellar/soroban-tools/issues/844

Integration tests won't pass until I update soroban-rpc and push a container with the update.

### CODE REVIEW

**Comment by 2opremio** [comment]
Soroban-rpc has been bump with the changes at https://github.com/stellar/soroban-tools/pull/846

However, a (puzzling hack) is needed for the sac tests to pass.

**Comment by 2opremio** [comment]
It seems to work if we ensure soroban-rpc is one ledger ahead of horizon when computing preflights.

It *doesn't* work if soroban-rpc is at the same ledger as horizon, which is puzzling.

**Comment by 2opremio** [comment]
The tests seem to be flaky even after the syncing between soroban-rpc and horizon: https://github.com/stellar/go/actions/runs/5791031972/job/15695140167?pr=5009

**Comment by 2opremio** [comment]
This was the cause of the problem state above https://github.com/stellar/soroban-tools/pull/846/commits/dcb79ce3eac989c4b7440935c9ff97717781b3d8

File: services/horizon/internal/test/integration/integration.go [hunk 1]
```diff
@@ -559,20 +559,26 @@ type RPCSimulateTxResponse struct {
...
+		var decodedRes xdr.ScVal
+		err := xdr.SafeUnmarshalBase64(res.XDR, &decodedRes)
+		assert.NoError(i.t, err)
+		fmt.Printf("Result:\n\n%# +v\n\n", pretty.Formatter(decodedRes))
+		for _, authBase64 := range res.Auth {
 			var authEntry xdr.SorobanAuthorizationEntry
-			err := xdr.SafeUnmarshalBase64(authBase64, &authEntry)
+			err = xdr.SafeUnmarshalBase64(authBase64, &authEntry)
 			assert.NoError(i.t, err)
 			fmt.Printf("Auth:\n\n%# +v\n\n", pretty.Formatter(authEntry))
```
Comment (sreuland) [inline services/horizon/internal/test/integration/integration.go:583]: could probably remove this auth printout now that whole result including auths are printed just prior.

**Review by sreuland — APPROVED** [review]

File: services/horizon/internal/test/integration/integration.go (hunk 1 above)
Comment (2opremio) [reply services/horizon/internal/test/integration/integration.go:583]: I don't think Auths are printed, just the soroban transaction data.

File: services/horizon/internal/test/integration/integration.go [hunk 2]
```diff
@@ -605,9 +616,25 @@ func (i *Test) simulateTransaction(
...
+		ch := jhttp.NewChannel("http://localhost:"+strconv.Itoa(sorobanRPCPort), nil)
+		sorobanRPCClient := jrpc2.NewClient(ch, nil)
+		err := sorobanRPCClient.CallResult(context.Background(), "getLatestLedger", nil, &result)
+		assert.NoError(i.t, err)
+		if result.Sequence >= ledgerToWaitFor {
+			return
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	i.t.Fatal("Time out waiting for soroban-rpc to sync")
```
[the remark this answers is not in the document]
Comment (2opremio) [reply services/horizon/internal/test/integration/integration.go:636]: There is no failure (we make sure with `assert.NoError(i.t, err)`)

**Pull request merged**